data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/configuration/AggregateThresholdOptions.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.s3.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.opensearch.dataprepper.model.types.ByteCount;
+
+
+/**
+ * An implementation class of s3 index configuration Options
+ */
+public class AggregateThresholdOptions {
+
+    private static final String DEFAULT_BYTE_CAPACITY = "50mb";
+
+    @JsonProperty("maximum_size")
+    private String maximumSize = DEFAULT_BYTE_CAPACITY;
```

Comment:
You can make this a `ByteCount` and data-prepper-core will parse it.

https://github.com/opensearch-project/data-prepper/blob/41f2d643cbad24cf29ca221abf25954f4b760d02/data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceConfig.java#L96-L97